src/services/postService.js

```diff
@@ -13,42 +13,50 @@ const {
   FETCH_TOP_X_POSTS,
   HUNDREDX_USERNAME,
   GET_BLOG_ENTRIES,
+  GET_CONTENT,
 }                  = require('../utils/constants');
 const VIDEO_URLS   = require('../utils/videoUrls');
 
 module.exports = {
+  broadcastAndStorePost: function({ authenticatedUserInstance, permLink, title, body, tags }) {
+    return authenticatedUserInstance.broadcastPost({ permLink, title, body, tags })
+      .then(broadcastSuccess => {
+        return broadcastSuccess
+          ? authenticatedUserInstance.userInOurDb
+          : null
+      })
+      .then(userRecord => {
+        return this.fetchSingleSteemitPost(permLink, userRecord);
+      })
+      .catch(err => {
+        new Error(err.error_description);
+      })
+  },
+
   reSyncPosts: function(posts, rankType) {
     for (const [tagIndex, postsByTag] of Object.entries(posts)) {
-      for (const [index, post] of Object.entries(postsByTag)) {
-        const newRanking = (tagIndex) * FETCH_POSTS_PER_TAG + parseInt(index);
-        const updateRankType = {};
-        updateRankType[rankType] = newRanking;
+      for (const [postIndex, steemitPost] of Object.entries(postsByTag)) {
+        // for each post from Steemit API...
+        const newRanking = this.calculateNewRank(tagIndex, postIndex);
+        const newRankingObj = this.generateProprietaryRankObject(rankType, newRanking);
+        const postForOurDb = this.formatSteemitPost(steemitPost);
+
         UserModel
           .findOrCreate({
-            where: {name: post.author},
-            defaults: { id: _.random(10000)}
+            where: {id: steemitPost.author},
```

Comment:
We should create a user with `id` and `name` field
```
User {
   id: 'someIdHash',
   name: 'name',
   etc: 'etc',
}
```

Reply:
Every table has an `id` column as the primary key

Reply:
Sounds good.  This has also been fixed in a seperate PR.  I will delete these changes when the merge conflicts come up.